src/base/memory_private_allocator.hpp

```diff
@@ -343,4 +343,26 @@ namespace cubmem
   }
 } // namespace cubmem
 
+template <typename Func, typename ...Args>
+inline void switch_to_global_allocator_and_call (Func && func, Args &&... args);
+
+template < typename Func, typename...Args >
+void
+switch_to_global_allocator_and_call (Func && func, Args && ... args)
+{
+#if defined(SERVER_MODE)
+  /* Switch to global context. */
+  HL_HEAPID save_id;
+
+  save_id = db_private_set_heapid_to_thread (NULL, 0);
```

Comment:
Use db_change_private_heap instead db_private_set_heapid_to_thread; should help with `#if defined` blocks.